src/com/can/CANLayer.cpp

```diff
@@ -58,26 +68,38 @@ EComResponse CCANComLayer::openConnection(char *paLayerParameter) {
     eRetVal = e_InitTerminated;
     break;
   case e_Publisher:
-    // is handled via sendData
+    // is only sendData
+    mFrame.can_id = can_params.CANId;
     break;
   case e_Subscriber:
-    // is handled via recvData
+    // is only recvData
+    rfilter.can_id = can_params.CANId;
+    rfilter.can_mask = CAN_SFF_MASK;
+    setsockopt(fd, SOL_CAN_RAW, CAN_RAW_FILTER, &rfilter, sizeof(rfilter));
     break;
-  }
 
-  DEVLOG_DEBUG("[CAN Layer] Opening connection\n");
+  default:
+    DEVLOG_ERROR("[CAN Layer] Invalid COM service type.\n");
```

Comment:
I would write "communication" instead of com

Reply:
For debugging I needed to see all ports in FB

Reply:
clean

Reply:
Update